### ISSUE TITLE
New compile system crashes site when rendering tab

### ISSUE THREAD

**Issue by larrywang0701** [source-academy/modules#181]
Hello
I found that in the new compile system, my code doesn't work and a runtime error crashed the whole site
`Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.`
The same code works well in old build system.

The code that may cause this crash is (100% tested well with old build system)

```tsx
class ProgrammableReplGUI extends React.Component<Props> {
(........)
  constructor(data: Props) {
    super(data);
(........)
  }
  public render() {
(........)
    return (
      <div>
(........)
      </div>
    );
  }
}
export default {
  toSpawn(_context: DebuggerContext) {
    return true;
  },
  body(context: DebuggerContext) {
    return <ProgrammableReplGUI programmableReplInstance={context.context.moduleContexts.repl.state} />;
  },
  label: 'Programmable Repl Tab',
  iconName: 'code',
};
```

**Comment by larrywang0701** [comment]
Full Stack Trace in Chrome console:
```
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of `ProgrammableReplGUI`.
    at createFiberFromTypeAndProps (react-dom.development.js:25058:1)
    at createFiberFromElement (react-dom.development.js:25086:1)
    at createChild (react-dom.development.js:13446:1)
    at reconcileChildrenArray (react-dom.development.js:13719:1)
    at reconcileChildFibers (react-dom.development.js:14125:1)
    at reconcileChildren (react-dom.development.js:16990:1)
    at updateHostComponent (react-dom.development.js:17632:1)
    at beginWork (react-dom.development.js:19080:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994:1)
    at invokeGuardedCallback (react-dom.development.js:4056:1)
    at beginWork$1 (react-dom.development.js:23964:1)
    at performUnitOfWork (react-dom.development.js:22776:1)
    at workLoopSync (react-dom.development.js:22707:1)
    at renderRootSync (react-dom.development.js:22670:1)
    at performSyncWorkOnRoot (react-dom.development.js:22293:1)
    at react-dom.development.js:11327:1
    at unstable_runWithPriority (scheduler.development.js:468:1)
    at runWithPriority$1 (react-dom.development.js:11276:1)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11322:1)
    at flushSyncCallbackQueue (react-dom.development.js:11309:1)
    at batchedUpdates$1 (react-dom.development.js:22387:1)
    at Object.notify (Subscription.js:16:1)
    at Object.notifyNestedSubs (Subscription.js:88:1)
    at handleChangeWrapper (Subscription.js:93:1)
    at dispatch (redux.js:297:1)
    at redux-saga-core.esm.js:1410:1
    at dispatch (redux.js:691:1)
    at io-6de156f3.js:133:1
    at redux-saga-core.esm.js:472:1
    at exec (redux-saga-core.esm.js:31:1)
    at flush (redux-saga-core.esm.js:87:1)
    at asap (redux-saga-core.esm.js:46:1)
    at runPutEffect (redux-saga-core.esm.js:468:1)
    at runEffect (redux-saga-core.esm.js:1204:1)
    at digestEffect (redux-saga-core.esm.js:1271:1)
    at next (redux-saga-core.esm.js:1161:1)
    at proc (redux-saga-core.esm.js:1108:1)
    at runCallEffect (redux-saga-core.esm.js:533:1)
    at runEffect (redux-saga-core.esm.js:1204:1)
    at digestEffect (redux-saga-core.esm.js:1271:1)
    at next (redux-saga-core.esm.js:1161:1)
    at currCb (redux-saga-core.esm.js:1251:1)
    at chCbAtKey (redux-saga-core.esm.js:698:1)
    at currCb (redux-saga-core.esm.js:1251:1)
```

**Comment by leeyi45** [comment]
Resolved with #183.
